OC/config.plist

```diff
@@ -227,14 +227,40 @@
 	<dict>
 		<key>Quirks</key>
```

Comment:
actually i also struggle for this too. 

cause:
- Kaby lake G is refreshed Kaby lake, and it's marketed as 8th Gen, 
- the guide is reference to 7th gen. 
- should we follow 7th gen or 8th gen

currently i am running on this configuration (8th gen) 
- i didn't notice any defects yet
- any power management issues, sleep
- kernel panics

this is worth digging on to it.

Reply:
This depends on MAT presence. Check the debug log if MAT is present, if it is, EnableWriteUnprotector should not be needed. Refer to Configuration.pdf for more details. Also make sure to try booting Windows and Linux before merging this. Setting the quirks incorrectly may result in breaking any of the systems.

Reply:
DevirtualiseMmio is safe to enable on any AMI firmware starting from Skylake, it allows you to reclaim a bit more memory in macOS.

Reply:
EnableSafeModeSlide depends on slide=0 availability on your target system. Easiest to check is through the debug log. You can also test safe mode (hold shift or use -x boot argument) with and without this quirk.

Reply:
SetupVirtualMap depends on the firmware implementation. If you end booting in any operating system with it disabled, it is not needed.

Reply:
> This depends on MAT presence. Check the debug log if MAT is present, if it is, EnableWriteUnprotector should not be needed. Refer to Configuration.pdf for more details. Also make sure to try booting Windows and Linux before merging this. Setting the quirks incorrectly may result in breaking any of the systems.

fragment of the log
```
01:811 00:036 OCABC: Got rendezvous with OpenRuntime r11
01:874 00:063 OCABC: MAT support is 1
01:971 00:097 OC: Driver OpenRuntime.efi at 0 is successfully loaded!
```
this indicate we need to enable EnableWriteUnprotector ?

Reply:
> EnableSafeModeSlide depends on slide=0 availability on your target system. Easiest to check is through the debug log. You can also test safe mode (hold shift or use -x boot argument) with and without this quirk.

tested safe mode with the quirk disabled, able to boot with success

Reply:
MAT 1 means that it is supported. Therefore you should not use EnableWriteUnprotector, but instead use RebuildAppleMemoryMap, SyncRuntimePermissions, and ProtectUefiServices.

Reply:
> MAT 1 means that it is supported. Therefore you should not use EnableWriteUnprotector, but instead use RebuildAppleMemoryMap, SyncRuntimePermissions, and ProtectUefiServices.

based on what you mention 

the Booter quirks is very closed to this commit  https://github.com/osy/HaC-Mini/pull/470/commits/d5e8fcc3055ed41735cb01f78169e65375f3fd85

Reply:
> MAT 1 means that it is supported. Therefore you should not use EnableWriteUnprotector, but instead use RebuildAppleMemoryMap, SyncRuntimePermissions, and ProtectUefiServices.

Dear @vit9696 may I ask you if `ProtectUefiServices` is needed even if without it I can successfully boot? 
Many thanks ^^

Reply:
Check whether MMIO devirtualisation happens in the log with and without this flag. If it disappears when you disable it, it is needed.

Reply:
> Check whether MMIO devirtualisation happens in the log with and without this flag. If it disappears when you disable it, it is needed.

Thank you so much for your reply. Well, I'm currently using `DevirtualiseMMIO` since without it I can't even boot with MAT Support 1 quirks. 

My current enabled quirks are:

- `AvoidRuntimeDefrag`
- `DevirtualiseMmio`
- `RebuildAppleMemoryMap`
- `SetupVirtualMap`
- `SyncRuntimePermissions`

Here you are the log file without ProtectUefiServices: [LOG](https://github.com/dreamwhite/dell-inspiron-5370-hackintosh/blob/master/SysReport/LOG%20MMIO%20opencore-2020-12-10-164336.txt#L466-L472)

> If it disappears when you disable it, it is needed.

So theorically I should leave it disabled, right?

Thank you so much ^^

Reply:
Yes, you _can_ leave it disabled. It will not do harm if it is enabled, but there is no real benefit either.

Reply:
> Yes, you can leave it disabled. It will not do harm if it is enabled, but there is no real benefit either.

Okay thank you so much. At this point I have a doubt: when should we enable DevirtualiseMMIO? Excluding situations like mine where I can't even boot without it

Reply:
Whenever it finds anything and it boots with it.